build.mill

```diff
@@ -0,0 +1,435 @@
+          linkLibraries().flatMap(l => Seq("-l", l))
+      ).call(Task.dest)
+      // @formatter:on
+      Seq(PathRef(Task.dest))
+    }
+  }
+
+  override def javacOptions = Task(super.javacOptions() ++ Seq("--enable-preview", "--release", "21"))
+}
+
+trait HasCIRCTPanamaBindingModule extends JavaModule {
+  import $file.circtpanamabinding.CIRCTPanamaBinding
+  def circtPanamaBindingModule: CIRCTPanamaBinding
+
+  override def moduleDeps = super.moduleDeps ++ Some(circtPanamaBindingModule)
+  //
+  override def javacOptions = Task(super.javacOptions() ++ Seq("--enable-preview", "--release", "21"))
+
+  override def forkArgs: T[Seq[String]] = Task(
+    super.forkArgs() ++ Seq("--enable-native-access=ALL-UNNAMED", "--enable-preview")
+      ++ circtPanamaBindingModule
+        .libraryPaths()
+        .map(p => s"-Djava.library.path=${p.path}")
+  )
+}
+
+trait HasPanamaLibModule extends ScalaModule with HasCIRCTPanamaBindingModule {
+  import $file.panamalib.PanamaLib
+  def panamaLibModule: PanamaLib
+
+  def circtPanamaBindingModule = panamaLibModule.circtPanamaBindingModule
+
+  override def moduleDeps = super.moduleDeps ++ Some(panamaLibModule)
+}
+
+trait HasPanamaOMModule extends ScalaModule with HasCIRCTPanamaBindingModule {
+  import $file.panamaom.PanamaOM
+  def panamaOMModule: PanamaOM
+
+  def circtPanamaBindingModule = panamaOMModule.circtPanamaBindingModule
+
+  override def moduleDeps = super.moduleDeps ++ Some(panamaOMModule)
+}
+
+trait HasPanamaConverterModule extends ScalaModule with HasCIRCTPanamaBindingModule {
+  import $file.panamaconverter.PanamaConverter
+  def panamaConverterModule: PanamaConverter
+
+  def circtPanamaBindingModule = panamaConverterModule.circtPanamaBindingModule
+
+  override def moduleDeps = super.moduleDeps ++ Some(panamaConverterModule)
+}
+
+object chisel extends Cross[Chisel](v.scalaCrossVersions)
+
+trait Chisel extends CrossSbtModule with HasScala2MacroAnno with HasScala2Plugin with ScalafmtModule {
+  override def millSourcePath = super.millSourcePath / os.up
+  def svsimModule = svsim.cross(crossScalaVersion)
+  def coreModule = core.cross(crossScalaVersion)
+  def pluginModule = plugin.cross()
+
+  override def scalacOptions = Task {
+    if (v.isScala3(crossScalaVersion)) {
+      Seq.empty[String]
+    } else {
+      super.scalacOptions() ++ v.scala2CommonOptions
+    }
+  }
+
+  override def moduleDeps = super.moduleDeps ++ Seq(coreModule, svsimModule)
+
+  object test extends SbtTests with TestModule.ScalaTest with ScalafmtModule {
+    def ivyDeps = Agg(v.scalatest, v.scalacheck)
+
+    // Suppress Scala 3 behavior requiring explicit types on implicit definitions
+    // Note this must come before the -Wconf is warningSuppression
+    override def scalacOptions = Task { super.scalacOptions() :+ "-Wconf:cat=other-implicit-type:s" }
+  }
+}
+
+/** Aggregate project for publishing Chisel as a single artifact
+  */
+object unipublish extends ScalaModule with ChiselPublishModule {
```

Comment:
I Personally think the `unipublish` is the only thing living in the `build.mill`, and external project should only depends on it.
If we wanna expose more things to external, we should goes to the publish flow to it, 
e.g. panamaom.

Reply:
An alternative is to split the publish flow into a separate file, as this can keep the `Has...` traits in `build.mill` to avoid a lot of `import build_($file)` in `package.mill`.

Reply:
A little tricky but fine to me. We can revisit it in the future when new mill feature released.